### ISSUE TITLE
imagep decimation shouldn't trigger warning with default options

### ISSUE THREAD

**Issue by richardsc** [dankelley/oce#845]
What I mean is, if someone calls `imagep()` with the default options (in which `decimate=TRUE`), the decimation shouldn't result in the warning about over decimating. That should be reserved for situations where the `decimate` argument is numeric -- i.e. specified explicitly by the user.

An example -- start with a matrix that has a very rectangular aspect ratio:
```r
library(oce)
mat <- matrix(rnorm(100*1000), nrow=1000)
imagep(mat)
Warning message:
In imagep(mat) : decimation leaves less than 100 pixels in height or width
```

I'm starting to wonder if perhaps `imagep()` should *always* warn on decimation, to avoid confusion about what's being plotted, especially when the default is to decimate. I mean, yes, everybody should RTFM, but I'm a package co-author, and before I got the above error I spent 10 minutes trying to figure out why an Argo matrix with 116 profiles was coming out looking like it had 20 (because it had 1000 pressure levels, so decimation was kicking in).

**Comment by dankelley** [comment]
Should decimate by direction..

**Comment by dankelley** [comment]
I feel this is working now in `develop`.  I did some tests with made-up matrices of different sizes, and also with different values of the `decimate` argument.  I now permit numerical versions of `decimate` to have two arguments, for the first and second index. Now, if `decimate=TRUE` and if it actually decimates, it will spit out a warning (or two, if it has to decimate in both directions).  The doc for the `decimate` arg is now broken down into cases, which I think should make things clearer for the user.

I think this is all working now. But it would be great to get an independent check, because `imagep` is both useful to users and complicated to program. I think having this bug fixed is a very high priority for the new release and I don't think we want to leave it to near the end of the test period.

I would hope to start a pre-release branch on Monday, if we have the basics of the new 'Next Release' labels handled.  But I don't think I want to do that until we have made good progress on these issues, because it might lead to a lot of confusion (for me) in merging the bug fixes into both `develop` and `prerelease`.
